packages/relic/lib/src/headers/headers.dart

```diff
@@ -0,0 +1,1688 @@
+        if (failedHeadersToParse.containsKey(key)) {
+          failedHeadersToParse[key]!.addAll(value);
+        } else {
+          failedHeadersToParse[key] = value;
+        }
+      },
+    );
+
+    return _HeadersImpl(
+      // Date-related headers
+      date: dartIOHeaders.parseDate(_dateHeader),
+      expires: dartIOHeaders.parseDate(_expiresHeader),
+      ifModifiedSince: dartIOHeaders.parseDate(_ifModifiedSinceHeader),
+      lastModified: dartIOHeaders.parseDate(_lastModifiedHeader),
+
+      // General Headers
+      origin: dartIOHeaders.parseUri(_originHeader),
+      server: dartIOHeaders.parseString(
+        _serverHeader,
+      ),
+      via: dartIOHeaders.parseStringList(
+        _viaHeader,
+      ),
+
+      // Request Headers
+      from: dartIOHeaders.parseMultipleValue(
+        _fromHeader,
+        onParse: FromHeader.parse,
+      ),
+      host: dartIOHeaders.parseUri(_hostHeader),
+      acceptEncoding: dartIOHeaders.parseMultipleValue(
+        _acceptEncodingHeader,
+        onParse: AcceptEncodingHeader.parse,
+      ),
+      acceptLanguage: dartIOHeaders.parseMultipleValue(
+        _acceptLanguageHeader,
+        onParse: AcceptLanguageHeader.parse,
+      ),
+      accessControlRequestHeaders: dartIOHeaders.parseStringList(
+        _accessControlRequestHeadersHeader,
+      ),
+      accessControlRequestMethod: dartIOHeaders.parseSingleValue(
+        _accessControlRequestMethodHeader,
+        onParse: Method.parse,
+      ),
+      age: dartIOHeaders.parseInt(
+        _ageHeader,
+        allowNegative: false,
+      ),
+      allow: dartIOHeaders.parseMultipleValue(
+        _allowHeader,
+        onParse: (values) {
+          return values
+              .splitTrimAndFilterUnique(emptyCheck: false)
+              .map(Method.parse)
+              .toList();
+        },
+      ),
+      cookie: dartIOHeaders.parseSingleValue(
+        _cookieHeader,
+        onParse: CookieHeader.parse,
+      ),
+      setCookie: dartIOHeaders.parseSingleValue(
+        _setCookieHeader,
+        onParse: CookieHeader.parse,
+      ),
+      authorization: dartIOHeaders.parseSingleValue(
+        _authorizationHeader,
+        onParse: AuthorizationHeader.parse,
+      ),
+      connection: dartIOHeaders.parseMultipleValue(
+        _connectionHeader,
+        onParse: ConnectionHeader.parse,
+      ),
+      expect: dartIOHeaders.parseSingleValue(
+        _expectHeader,
+        onParse: ExpectHeader.parse,
+      ),
+      ifMatch: dartIOHeaders.parseMultipleValue(
+        _ifMatchHeader,
+        onParse: IfMatchHeader.parse,
+      ),
+      ifNoneMatch: dartIOHeaders.parseMultipleValue(
+        _ifNoneMatchHeader,
+        onParse: IfNoneMatchHeader.parse,
+      ),
+      ifRange: dartIOHeaders.parseSingleValue(
+        _ifRangeHeader,
+        onParse: IfRangeHeader.parse,
+      ),
+      maxForwards: dartIOHeaders.parseInt(
+        _maxForwardsHeader,
+        allowNegative: false,
+      ),
```

Comment:
Suggestion: Create a parser called `parsePositiveInt` instead of passing in bools from this place.